lib/ansible/modules/cloud/azure/azure_rm_managed_disk.py

```diff
@@ -327,7 +342,9 @@ def attach(self, vm_name, disk):
 
         # prepare the data disk
         params = self.compute_models.ManagedDiskParameters(id=disk.get('id'), storage_account_type=disk.get('storage_account_type'))
-        data_disk = self.compute_models.DataDisk(lun=lun, create_option=self.compute_models.DiskCreateOptionTypes.attach, managed_disk=params)
+        caching_options = self.compute_models.CachingTypes[self.caching] if self.caching and self.caching != '' else None
+        data_disk = self.compute_models.DataDisk(lun=lun, create_option=self.compute_models.DiskCreateOptionTypes.attach,
```

Comment:
*For linting:* If here need to split into multi lines, we prefer one option one line

Reply:
In my understanding, the function ```is_different(self, found_disk, new_disk)``` is used to find the different properties between the ```found_disk``` and ```new_disk```. However, caching option is not a property of the disk.